lib/cvss.js

```diff
@@ -161,7 +161,83 @@ function CVSS(vector) {
     return definitions.definitions.find((def) => def.abbr === abbr);
   };
 
-  return { vector, getScore, getRating, getVectorObject, getDetailedVectorObject, getTemporalScore };
+  /**
+   * Checks whether the vector passed is valid
+   *
+   * @returns {{message: string, isValid: boolean}} object with "message" and "isValid"
+   */
+  const isVectorValid = function () {
+    /**
+     * This function is used to scan the definitions file and join all
+     * abbreviations in a format that RegExp understands.
+     * 
+     * Exit example:
+     * ((((((((((AV:[NALP]|AC:[LH])|PR:[NLH])|UI:[NR])|S:[UC])|C:[NLW])|I:[NLW])|A:[NLW])|E:[XUPFH])|RL:[XOTWU])|RC:[XURC])
+     */
+    const expression = definitions.definitions.reduce((accumulator, currentValue, index) => {
+      const serializedAbbr = `${currentValue.abbr}:[${currentValue.metrics.reduce((accumulator2, currentValue2) => {
+        return accumulator2 + currentValue2.abbr;
+      }, "")}]`;
+      if (index !== 0) {
+        return `(${accumulator }|${serializedAbbr})`;
+      } else {
+        return serializedAbbr;
+      }
+    }, "");
+
+    const totalExpressionVector = new RegExp("^CVSS:3.0(/" + expression + ")+$");
+
+    //Checks if the vector is in valid format
+    if(!totalExpressionVector.test(vector)){
+      return {
```

Comment:
I'm not sure if we should return an object here instead of just true or false. I think in a next step it would be good if the vector is already validated when it is created and if it is invalid, an exception is thrown. In that way we make sure that our vector is always valid and the user can not call the other methods if the vector is invalid. This would save us the effort of securing the other functions, resp. checking in each function if the vector is valid.

Reply:
ok, my idea with the object was to inform the user why the vector is invalid

Reply:
So, should I add the initial check to the role when it is created?

Reply:
I think the best idea would be to throw an Error if the user calls `CVSS()` with an invalid vector. Maybe we could set a "class-wide" variable indicating if the vector is valid. I do not know how useful this would be, because isValid can never be false.

So for example:
Valid vectors:
```js
const vector = CVSS("CVSS:3.0/AV:L/AC:H/PR:L/UI:R/S:C/C:L/I:L/A:L/E:U/RL:T/RC:R")
console.log(vector.isValid) // true
```

And for invalid vectors
```js
const vector = CVSS("CVSS:3.0/AV:N/AC:H/PR:L/UI:R/S:C/C:L/I:L/A:L/E:U/RL:T/RC:R/RC:R"); // Throws Error with message
```

Reply:
I think it will be useful only for testing purposes and as a guarantee to the user that everything is correct.

Reply:
I'll change the code right now, thanks!

Reply:
the isVectorValid function still needs to be public if it will now have the variable isValid?

Reply:
i think you can just use a variable in the "root" CVSS function and provide it to the user, right?

Reply:
What I mean is something like
`const isValid = isVectorValid()`

and then provide `isValid` in the object that is returned.

Reply:
Yes, I have another question, can I run the tests with an error being launched?

Reply:
Yes, I think you can use `expect(CVSS("VECTOR")).toThrowError("ERROR_MESSAGE")`

https://jestjs.io/docs/en/expect#tothrowerror

Reply:
Thank you!